test/operations/wait.go

```diff
@@ -70,14 +70,20 @@ func Poll(interval, deadline time.Duration, condition ConditionFunc) error {
 type ethClienter interface {
 	ethereum.TransactionReader
 	ethereum.ContractCaller
+	bind.DeployBackend
 }
 
 // WaitTxToBeMined waits until a tx has been mined or the given timeout expires.
-func WaitTxToBeMined(client ethClienter, hash common.Hash, timeout time.Duration) error {
-	ctx := context.Background()
-	return Poll(DefaultInterval, timeout, func() (bool, error) {
-		return txMinedCondition(ctx, client, hash)
-	})
+func WaitTxToBeMined(parentCtx context.Context, client ethClienter, tx *types.Transaction, timeout time.Duration) error {
+	ctx, cancel := context.WithTimeout(parentCtx, timeout)
+	defer cancel()
+	_, err := bind.WaitMined(ctx, client, tx)
+	if err != nil {
+		log.Error("error waiting tx to be mined: ", err)
```

Comment:
```suggestion
		log.Errorf("error waiting tx %s to be mined: %w", tx.Hash(), err)
```

Reply:
Sometimes, `tx` can be `nil`, in that case, it will break the entire application